uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/services/task.js

```diff
@@ -1,271 +1,294 @@
-                        return null;
-                    }
-
-                    var deferred = $q.defer();
-
-                    Task.save({
-                        action: 'sendreminder',
-                        json: {
-                            sequential: 1,
-                            debug: config.DEBUG,
-                            activity_id: taskId,
-                            notes: notes || ''
-                        }
-                    }, null, function (data) {
-
-                        if (UtilsService.errorHandler(data, 'Unable to send a reminder', deferred)) {
-                            return;
-                        }
-
-                        deferred.resolve(data);
-                    }, function () {
-                        deferred.reject('Unable to send a reminder');
-                    });
-
-                    return deferred.promise;
-                },
-                delete: function (taskId) {
-
-                    if (!taskId || typeof +taskId !== 'number') {
-                        return null;
-                    }
-
-                    var deferred = $q.defer();
-
-                    Task.delete({
-                        action: 'delete',
-                        json: { id: taskId }
-                    }, function (data) {
-                        deferred.resolve(data);
-                    }, function () {
-                        deferred.reject('Could not delete task ID: ' + taskId);
-                    });
-
-                    return deferred.promise;
-                }
+  'common/angular',
+  'common/moment',
+  'tasks-assignments/services/services',
+  'tasks-assignments/services/utils'
+], function(angular, moment, services) {
+  'use strict';
+
+  services.factory('Task', ['$resource', '$httpParamSerializer', 'config', '$log', function($resource, $httpParamSerializer, config, $log) {
+    $log.debug('Service: Task');
+
+    function transformUrlEncoded(data) {
+      return $httpParamSerializer(data);
+    }
+
+    return $resource(config.url.REST, {
+      'action': 'get',
+      'entity': 'Task'
+    }, {
+      save: {
+        method: "POST",
+        isArray: false,
+        headers: {
+          'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8'
+        },
+        transformRequest: transformUrlEncoded
+      }
+    });
+  }]);
+
+  services.factory('TaskService', ['Task', '$q', 'config', 'UtilsService', '$log',
+    function(Task, $q, config, UtilsService, $log) {
+      $log.debug('Service: TaskService');
+
+      return {
+        assign: function(taskArr, assignmentId) {
+
+          if ((!taskArr || !angular.isArray(taskArr)) ||
+            (!assignmentId || typeof + assignmentId !== 'number')) {
+            return null;
+          }
+
+          if (!taskArr.length) {
+            return taskArr;
+          }
+
+          var deferred = $q.defer();
+
+          Task.save({}, {
+            action: 'copy_to_assignment',
```

Comment:
Now you're passing *all* the parameters in the request body, but i don't think that's necessary? Shouldn't you just pass `json` in the body?